packages/react-native-worklets/src/animationFrameQueue/requestAnimationFrame.ts

```diff
@@ -4,38 +4,64 @@ import { callMicrotasks } from '../threads';
 
 export function setupRequestAnimationFrame() {
   'worklet';
+  const nativeRequestAnimationFrame = globalThis.requestAnimationFrame;
 
-  // Jest mocks requestAnimationFrame API and it does not like if that mock gets overridden
-  // so we avoid doing requestAnimationFrame batching in Jest environment.
-  const nativeRequestAnimationFrame = global.requestAnimationFrame;
+  let animationFrameCallbacks: ((timestamp: number) => void)[] = [];
+  let callbacksBegin = 0;
+  let callbacksEnd = 0;
+
+  let flushedCallbacks = animationFrameCallbacks;
+  let flushedCallbacksBegin = 0;
+  let flushedCallbacksEnd = 0;
 
-  let animationFrameCallbacks: Array<(timestamp: number) => void> = [];
   let flushRequested = false;
 
-  global.__flushAnimationFrame = (frameTimestamp: number) => {
-    const currentCallbacks = animationFrameCallbacks;
+  globalThis.__flushAnimationFrame = (timestamp: number) => {
+    globalThis.__frameTimestamp = timestamp;
+
+    flushedCallbacks = animationFrameCallbacks;
     animationFrameCallbacks = [];
-    currentCallbacks.forEach((f) => f(frameTimestamp));
+
+    flushedCallbacksBegin = callbacksBegin;
+    flushedCallbacksEnd = callbacksEnd;
+    callbacksBegin = callbacksEnd;
+
+    flushRequested = false;
+
+    for (const callback of flushedCallbacks) {
+      callback(timestamp);
+    }
+
+    flushedCallbacksBegin = callbacksEnd;
```

Comment:
It would be more intuitive if you use the variable family related to the same callback set.

```suggestion
    flushedCallbacksBegin = flushedCallbacksEnd;
```